modes/longitudinal/src/toolbarButtons.js

```diff
@@ -295,26 +296,41 @@ const toolbarButtons = [
     },
   },
   {
-    id: 'Layout',
-    type: 'ohif.layoutSelector',
-    props: {
-      rows: 3,
-      columns: 3,
-    },
-  },
-  {
-    id: 'MPR',
-    type: 'ohif.action',
+    id: 'Layouts',
+    type: 'ohif.splitButton',
     props: {
-      type: 'toggle',
-      icon: 'icon-mpr',
-      label: 'MPR',
-      commands: [
-        {
-          commandName: 'toggleMPR',
-          commandOptions: {},
-          context: 'CORNERSTONE',
+      isRadio: false,
+      groupId: 'Layouts',
+      primary: {
+        id: 'Layout',
+        type: 'action',
+        uiType: 'ohif.layoutSelector',
```

Comment:
Here you can add a command list of things to execute when the layout is toggled.  The first item should be one to
turn off any toggled layouts (eg toggleMPR for now), while the second should be a new command that simply calls setLayout(rows,columns) after getting rows/columns from the arguments.
That allows in the future to do things like have custom layouts configured in the hanging protocol and switching to those - for example, a 1x1 layout might just "toggle" the viewport to a single volume view 1x1 rather than  switching away from MPR mode.

Reply:
See previous comment [regarding StateSyncService.](https://github.com/OHIF/Viewers/pull/3086#discussion_r1062566151)

Reply:
OK, it can go in a separate PR - I agree, the general change layout solution needs to be thought out more carefully.

Reply:
Ok. I added a bunch of TODOs. In essence, I agree with you. Changing grid layout amounts to: saving the current app/HP state; switching to some hanging protocol (likely default) and then changing to the grid selected.  Currently the toggleMPR command does the state saving and restoring, but it shouldn't.

Reply:
We can discuss this further, but I think maybe in the next PR the HangingProtocolService should (perhaps via another service) save the current app/hp state when switching to another hp/stage. Furthermore when switching back to a previously used hp/stage, that previous state should be restored.